src/test/java/com/proofpoint/discovery/InMemoryDynamicStore.java

```diff
@@ -99,39 +101,59 @@ public synchronized void delete(Id<Node> nodeId)
                 .filter(matchesType(type).and(matchesPool(pool)));
     }
 
+    @Override
+    public Entry get(Id<Node> nodeId)
+    {
+        removeExpired();
```

Comment:
I'm concerned about doubling the expiration scans.

Perhaps replace this with an expiration check on `localEntry`?